xarray/core/netcdftimeindex.py

```diff
@@ -0,0 +1,180 @@
+import re
+from datetime import timedelta
+
+import numpy as np
+import pandas as pd
+
+from pandas.lib import isscalar
+
+
+def named(name, pattern):
+    return '(?P<' + name + '>' + pattern + ')'
+
+
+def optional(x):
+    return '(?:' + x + ')?'
+
+
+def trailing_optional(xs):
+    if not xs:
+        return ''
+    return xs[0] + optional(trailing_optional(xs[1:]))
+
+
+def build_pattern(date_sep='\-', datetime_sep='T', time_sep='\:'):
+    pieces = [(None, 'year', '\d{4}'),
```

Comment:
Personally, I don't see myself needing it in the near future, but I'm not necessarily opposed to adding that support if others would find it useful.

It would make writing simple positive four-digit year dates more complicated though right?  Would you always need the leading zero and the sign?